src/connectedk8s/azext_connectedk8s/_params.py

```diff
@@ -5,10 +5,13 @@
 # pylint: disable=line-too-long
 
 from argcomplete.completers import FilesCompleter
-from azure.cli.core.commands.parameters import get_location_type
+from azure.cli.core.commands.parameters import get_location_type, get_enum_type
 from azure.cli.core.commands.parameters import (file_type)
 from azure.cli.core.commands.validators import get_default_location_from_resource_group
 
+distribution_types = ["auto", "generic", "openshift", "rancher_RKE", "kind", "GKE", "EKS", "AKS", "AKS-HCI", "CAPZ", "AKS-engine", "TKG"]
```

Comment:
Use following values:

- Azure Kubernetes Service
- Google Kubernetes Engine
- Amazon Elastic Kubernetes Service
- Rancher Kubernetes Engine
- OpenShift Kubernetes Distribution
- OpenShift Container Platform
- K3s
- kind
- minikube
- Cluster API Provider Azure
- AKS Engine
- Tanzu Kubernetes Grid
- Canonical Charmed Kubernetes

Reply:
We should check if we can distinguish between OKD and OCP in above list.

Reply:
We should also check if Docker for [Mac](https://docs.docker.com/docker-for-mac/#kubernetes)/[Windows](https://docs.docker.com/docker-for-windows/#kubernetes) has any cluster identifying characteristic in which case we can include that in the list as well